osu.Game/Screens/Edit/Editor.cs

```diff
@@ -703,11 +712,53 @@ private List<MenuItem> createFileMenuItems()
             if (RuntimeInfo.IsDesktop)
                 fileMenuItems.Add(new EditorMenuItem("Export package", MenuItemType.Standard, exportBeatmap));
 
+            fileMenuItems.Add(new EditorMenuItemSpacer());
+
+            var beatmapSet = beatmapManager.QueryBeatmapSet(bs => bs.ID == Beatmap.Value.BeatmapSetInfo.ID) ?? playableBeatmap.BeatmapInfo.BeatmapSet;
+
+            var difficultyItems = new List<MenuItem>();
+
+            foreach (var rulesetBeatmaps in beatmapSet.Beatmaps.GroupBy(b => b.RulesetID).OrderBy(group => group.Key))
+            {
+                if (difficultyItems.Count > 0)
+                    difficultyItems.Add(new EditorMenuItemSpacer());
+
+                foreach (var beatmap in rulesetBeatmaps.OrderBy(b => b.StarDifficulty))
+                    difficultyItems.Add(createDifficultyMenuItem(beatmap));
+            }
+
+            fileMenuItems.Add(new EditorMenuItem("Change difficulty") { Items = difficultyItems });
+
             fileMenuItems.Add(new EditorMenuItemSpacer());
             fileMenuItems.Add(new EditorMenuItem("Exit", MenuItemType.Standard, this.Exit));
             return fileMenuItems;
         }
 
+        private DifficultyMenuItem createDifficultyMenuItem(BeatmapInfo beatmapInfo)
+        {
+            bool isCurrentDifficulty = playableBeatmap.BeatmapInfo.Equals(beatmapInfo);
+            return new DifficultyMenuItem(beatmapInfo, isCurrentDifficulty, switchToDifficulty);
+        }
+
+        private void switchToDifficulty(BeatmapInfo beatmapInfo)
+        {
+            if (loader == null)
+                return;
+
+            loader.ValidForResume = true;
+            this.Exit();
+            loader.ScheduleDifficultySwitch(beatmapInfo);
+        }
+
+        private void cancelPendingDifficultySwitch()
+        {
+            if (loader == null)
+                return;
+
+            loader.ValidForResume = false;
+            loader.CancelDifficultySwitch();
```

Comment:
Same here, I find it weird to set states onto another screen before calling something on that screen that makes use of the state.